File: 3144/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(minimumSubstringsInPartition("fabccddg")) // 3
	loop()
}

func loop() {
	c := 0
Loop:
	for c < 10 {
		if c == 5 {
			continue Loop
		}
		c++
		print(c)
	}
	print(c)
}

func minimumSubstringsInPartition(s string) int {
	n := len(s)
	d := make([]int, n+1)
	for i := range d {
		d[i] = 0x3f3f3f3f
	}
	d[0] = 0
	for i := 1; i <= n; i++ {
		maxCnt := 0
		occCnt := make(map[byte]int)
		for j := i; j >= 1; j-- {
			occCnt[s[j-1]]++
			if occCnt[s[j-1]] > maxCnt {
				maxCnt = occCnt[s[j-1]]
			}
			if maxCnt * len(occCnt) == (i-j+1) &&d[j-1] != 0x3f3f3f3f {
				if d[i] > d[j-1]+1 {
					d[i] = d[j-1]+1
				}
			}
		}
	}
	return d[n]
}

File: 3144/main.rs
const INF: i32 = 0x3f3f3f3f;

struct Solution {}

impl Solution {
    pub fn minimum_substrings_in_partition(s: String) -> i32 {
        let n = s.len();
        let mut d = vec![INF; n+1];
        d[0] = 0;
        for i in 1..=n {
            let mut max_cnt = 0;
            let mut occ_cnt = std::collections::HashMap::new();
            for j in (1..=i).rev() {
                let c = s.as_bytes()[j-1];
                *occ_cnt.entry(c).or_insert(0) += 1;
                max_cnt = max_cnt.max(*occ_cnt.get(&c).unwrap());
                if max_cnt * occ_cnt.len() == i-j+1 && d[j-1] != INF {
                    d[i] = d[i].min(d[j-1]+1);
                }
            }
        }
        d[n]
    }
}

fn main() {
    println!("{}", Solution::minimum_substrings_in_partition("fabccddg".to_string()));  // 3
}